x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -0,0 +1,565 @@
+                            StandardCopyOption.COPY_ATTRIBUTES);
+                } else {
+                    Files.deleteIfExists(keystorePath);
+                }
+            } catch (Exception ex) {
+                e.addSuppressed(ex);
+            }
+            try {
+                Files.deleteIfExists(instantAutoConfigDir);
+            } catch (Exception ex) {
+                e.addSuppressed(ex);
+            }
+            if (false == (e instanceof UserException)) {
+                throw e; // unexpected exections should prevent the node from starting
+            }
+            if (options.has(strictOption)) {
+                throw e;
+            } else {
+                return; // ignoring if the keystore contains password values already, so that the node startup deals with it (fails)
+            }
+        } finally {
+            if (nodeKeystorePassword.get() != null) {
+                nodeKeystorePassword.get().close();
+            }
+        }
+
+        try {
+            List<String> existingConfigLines = Files.readAllLines(ymlPath, StandardCharsets.UTF_8);
+            fullyWriteFile(env.configFile(), "elasticsearch.yml", true, stream -> {
+                try (BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(stream, StandardCharsets.UTF_8))) {
+                    // start with the existing config lines
+                    for (String line : existingConfigLines) {
+                        bw.write(line);
+                        bw.newLine();
+                    }
+                    bw.newLine();
+                    bw.newLine();
+                    bw.write("###################################################################################");
+                    bw.newLine();
+                    bw.write("# The following settings, and associated TLS certificates and keys configuration, #");
+                    bw.newLine();
+                    bw.write("# have been automatically generated in order to configure Security.               #");
+                    bw.newLine();
+                    bw.write("# These have been generated the first time that the new node was started, without #");
+                    bw.newLine();
+                    bw.write("# joining or enrolling to an existing cluster and only if Security had not been   #");
+                    bw.newLine();
+                    bw.write("# explicitly configured beforehand.                                               #");
+                    bw.newLine();
+                    bw.write(String.format(Locale.ROOT, "# %-79s #", ""));
+                    bw.newLine();
+                    bw.write(String.format(Locale.ROOT, "# %-79s #", autoConfigDate));
+                    // TODO add link to docs
+                    bw.newLine();
+                    bw.write("###################################################################################");
```

Comment:
Don't we want to put an instruction here on how to recover (remove this section and enroll the node) if a new cluster was mistakenly configured instead of joining an existing one?